src/ray/gcs/redis_client.cc

```diff
@@ -120,7 +121,41 @@ static void GetRedisShards(redisContext *context,
   freeReplyObject(reply);
 }
 
-RedisClient::RedisClient(const RedisClientOptions &options) : options_(options) {}
+RedisClient::RedisClient(const RedisClientOptions &options) : options_(options) {
+  instrumented_io_context io_context;
+  auto tmp_context = std::make_shared<RedisContext>(io_context);
+  RAY_CHECK_OK(tmp_context->Connect(options_.server_ip_,
+                                    options_.server_port_,
+                                    /*sharding=*/options_.enable_sharding_conn_,
+                                    /*password=*/options_.password_,
+                                    /*enable_ssl=*/options_.enable_ssl_));
+  auto reply = tmp_context->RunArgvSync(std::vector<std::string>{"INFO", "REPLICATION"});
```

Comment:
does this logic also work with none replicated redis?

Reply:
yes, we have ci test covering this.